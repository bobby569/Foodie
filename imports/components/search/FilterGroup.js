import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Select } from 'antd';
const { Option } = Select;

class FilterGroup extends Component {
	render() {
		const { onDietChange, onHealthChange } = this.props;

		return (
			<div className="search-filter">
				<Row type="flex" justify="center">
					<Col span={12}>
						<Select
							className="search-select"
							defaultValue="intro"
							onChange={onDietChange}
						>
							<Option value="intro" disabled>
								Choose Diet
							</Option>
							<Option value="balanced">Balanced</Option>
							<Option value="high-fiber">High-Fiber</Option>
							<Option value="high-protein">High-Protein</Option>
							<Option value="low-carb">Low-Carb</Option>
							<Option value="low-fat">Low-Fat</Option>
							<Option value="low-sodium">Low-Sodium</Option>
							<Option value="none">No Restriction</Option>
						</Select>
					</Col>
					<Col span={12}>
						<Select
							className="search-select"
							defaultValue="intro"
							onChange={onHealthChange}
						>
							<Option value="intro" disabled>
								Choose Health
							</Option>
							<Option value="alcohol-free">Alcohol free</Option>
							<Option value="fish-free">Fish Free</Option>
							<Option value="gluten-free">Gluten Free</Option>
							<Option value="low-potassium">Low potassium</Option>
							<Option value="low-sugar">No sugar</Option>
							<Option value="peanut-free">Peanut Free</Option>
							<Option value="pork-free">Pork free</Option>
							<Option value="tree-nut-free">Tree Nut Free</Option>
							<Option value="soy-free">Soy Free</Option>
							<Option value="vegetarian">Vegetarian</Option>
							<Option value="none">No Restriction</Option>
						</Select>
					</Col>
				</Row>
			</div>
		);
	}
}

FilterGroup.propTypes = {
	onDietChange: PropTypes.func.isRequired,
	onHealthChange: PropTypes.func.isRequired
};

export default FilterGroup;
