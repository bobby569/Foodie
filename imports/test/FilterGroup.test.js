import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FilterGroup from '../components/search/FilterGroup';

const onDietChange = jest.fn();
const onHealthChange = jest.fn();

describe('FilterGroup', () => {
	const filter_group = shallow(
		<FilterGroup onDietChange={onDietChange} onHealthChange={onHealthChange} />
	);
	const rendered = renderer.create(
		<FilterGroup onDietChange={onDietChange} onHealthChange={onHealthChange} />
	);

	it('Renders correctly', () => {
		expect(filter_group.exists()).toBe(true);
		expect(rendered.toJSON()).toMatchSnapshot();
	});
});
