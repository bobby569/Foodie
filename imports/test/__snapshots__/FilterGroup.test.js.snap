// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterGroup Renders correctly 1`] = `
<div
  className="search-filter"
>
  <div
    className="ant-row-flex ant-row-flex-center"
    style={undefined}
  >
    <div
      className="ant-col-12"
    >
      <div
        className="search-select ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={undefined}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          aria-autocomplete="list"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="Choose Diet"
            >
              Choose Diet
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col-12"
    >
      <div
        className="search-select ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={undefined}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          aria-autocomplete="list"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="Choose Health"
            >
              Choose Health
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
